.github/workflows/build-knox-retriever-image.yml

```diff
@@ -0,0 +1,78 @@
+name: Build and Push Docker Image
+
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+    tags:
+      - "v*"
+env:
+  IMAGE_NAME: ghcr.io/${{ github.repository }}/nr-broker-openshift-knox-retriever
+
+jobs:
+  # Job for building and pushing the image for the test environment
+  build-and-push-test:
```

Comment:
Do we need to split test/prod? I'm unclear why this is needed.

Reply:
use only for prod